Mathlib/MeasureTheory/Integral/IntegrableOn.lean

```diff
@@ -158,41 +167,48 @@ theorem IntegrableOn.left_of_union (h : IntegrableOn f (s ∪ t) μ) : Integrabl
 theorem IntegrableOn.right_of_union (h : IntegrableOn f (s ∪ t) μ) : IntegrableOn f t μ :=
   h.mono_set subset_union_right
 
-theorem IntegrableOn.union (hs : IntegrableOn f s μ) (ht : IntegrableOn f t μ) :
+theorem IntegrableOn.union [PseudoMetrizableSpace ε]
+    (hs : IntegrableOn f s μ) (ht : IntegrableOn f t μ) :
     IntegrableOn f (s ∪ t) μ :=
   (hs.add_measure ht).mono_measure <| Measure.restrict_union_le _ _
 
 @[simp]
-theorem integrableOn_union : IntegrableOn f (s ∪ t) μ ↔ IntegrableOn f s μ ∧ IntegrableOn f t μ :=
+theorem integrableOn_union [PseudoMetrizableSpace ε] :
+    IntegrableOn f (s ∪ t) μ ↔ IntegrableOn f s μ ∧ IntegrableOn f t μ :=
   ⟨fun h => ⟨h.left_of_union, h.right_of_union⟩, fun h => h.1.union h.2⟩
 
 @[simp]
-theorem integrableOn_singleton_iff {x : α} [MeasurableSingletonClass α] :
+theorem integrableOn_singleton_iff {f : α → ε'} {x : α}
+    [MeasurableSingletonClass α] (hfx : ‖f x‖ₑ ≠ ⊤) :
     IntegrableOn f {x} μ ↔ f x = 0 ∨ μ {x} < ∞ := by
   have : f =ᵐ[μ.restrict {x}] fun _ => f x := by
     filter_upwards [ae_restrict_mem (measurableSet_singleton x)] with _ ha
     simp only [mem_singleton_iff.1 ha]
-  rw [IntegrableOn, integrable_congr this, integrable_const_iff, isFiniteMeasure_restrict,
-    lt_top_iff_ne_top]
+  rw [IntegrableOn, integrable_congr this, integrable_const_iff_enorm, isFiniteMeasure_restrict,
+    lt_top_iff_ne_top, enorm_eq_zero]
+  exact hfx
 
 @[simp]
-theorem integrableOn_finite_biUnion {s : Set β} (hs : s.Finite) {t : β → Set α} :
+theorem integrableOn_finite_biUnion [PseudoMetrizableSpace ε]
+    {s : Set β} (hs : s.Finite) {t : β → Set α} :
     IntegrableOn f (⋃ i ∈ s, t i) μ ↔ ∀ i ∈ s, IntegrableOn f (t i) μ := by
   induction s, hs using Set.Finite.induction_on with
   | empty => simp
   | insert _ _ hf => simp [hf, or_imp, forall_and]
 
 @[simp]
-theorem integrableOn_finset_iUnion {s : Finset β} {t : β → Set α} :
+theorem integrableOn_finset_iUnion [PseudoMetrizableSpace ε] {s : Finset β} {t : β → Set α} :
     IntegrableOn f (⋃ i ∈ s, t i) μ ↔ ∀ i ∈ s, IntegrableOn f (t i) μ :=
   integrableOn_finite_biUnion s.finite_toSet
 
 @[simp]
-theorem integrableOn_finite_iUnion [Finite β] {t : β → Set α} :
+theorem integrableOn_finite_iUnion [PseudoMetrizableSpace ε] [Finite β] {t : β → Set α} :
     IntegrableOn f (⋃ i, t i) μ ↔ ∀ i, IntegrableOn f (t i) μ := by
   cases nonempty_fintype β
-  simpa using @integrableOn_finset_iUnion _ _ _ _ _ f μ Finset.univ t
+  simpa using integrableOn_finset_iUnion (f := f) (μ := μ) (s := Finset.univ) (t := t)
 
+-- TODO: generalise this lemma and the next to enorm classes; this entails assuming that
+-- f is integrable on each singleton, i.e. finite almost everywhere...
```

Comment:
```suggestion
-- f is finite on each element of `s`
```
right?

Reply:
I'd think your phrasing is slightly weaker: if f is infinite on some s, but that element has measure zero --- we are fine, right?